src/expressionsem.c

```diff
@@ -1806,11 +1806,19 @@ class ExpressionSemanticVisitor : public Visitor
         Expression *e;
         if (ea && ta->toBasetype()->ty == Tclass)
         {
-            /* Get the dynamic type, which is .classinfo
-            */
-            ea = semantic(ea, sc);
-            e = new TypeidExp(ea->loc, ea);
-            e->type = Type::typeinfoclass->type;
+            if (!Type::typeinfoclass)
+            {
+                error(exp->loc, "`object.TypeInfo_Class` could not be found, but is implicitly used");
```

Comment:
I would include what is implicitly using `TypeInfo_Class` in the error message. That is, `typeid` or `classinfo`.

Reply:
The loc would give that away with contextual errors, feel free to change it in master.